### ISSUE TITLE
[REVIEW]: RECOLO: A Python package for the reconstruction of surface pressure loads from kinematic fields using the virtual fields method

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3980]
**Submitting author:** @PolymerGuy (<a href="http://orcid.org/0000-0002-9981-4041">Sindre Nordmark Olufsen</a>)
**Repository:** https://github.com/PolymerGuy/recolo
**Branch with paper.md** (empty if default branch): JOSS-review
**Version:** v1.0.1
**Editor:** @Kevin-Mattheus-Moerman
**Reviewers:** @nicoguaro, @corentin-dev
**Archive:** 10.5281/zenodo.6324870

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/9c0b1dc9e5e554516bcad0fb918f4708/status.svg)](https://joss.theoj.org/papers/9c0b1dc9e5e554516bcad0fb918f4708)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/9c0b1dc9e5e554516bcad0fb918f4708"><img src="https://joss.theoj.org/papers/9c0b1dc9e5e554516bcad0fb918f4708/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/9c0b1dc9e5e554516bcad0fb918f4708/status.svg)](https://joss.theoj.org/papers/9c0b1dc9e5e554516bcad0fb918f4708)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/PolymerGuy/recolo" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@nicoguaro & @corentin-dev, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @Kevin-Mattheus-Moerman know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @nicoguaro
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/PolymerGuy/recolo">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/PolymerGuy/recolo">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@PolymerGuy) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @corentin-dev
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/PolymerGuy/recolo">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/PolymerGuy/recolo">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@PolymerGuy) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @nicoguaro, @corentin-dev it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 589

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/2041419621993492 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.ijmecsci.2020.106263 is OK
- 10.1007/s00348-019-2850-y is OK
- 10.1007/s11340-019-00530-2 is OK
- 10.1177/2041419616666236 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.13 s (425.0 files/s, 35086.1 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          41            804            852           1928
reStructuredText                 6            141             61            248
TeX                              2             10              0            174
Markdown                         2             34              0            100
YAML                             1              9              9             32
DOS Batch                        1              8              1             27
make                             1              4              6             10
-------------------------------------------------------------------------------
SUM:                            54           1010            929           2519
-------------------------------------------------------------------------------


Statistical information for the repository '846dfe5ec6200ff6bc00219f' was
gathered on 2021/12/07.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Sindre Nordmark Oluf           181         12494           8915          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Sindre Nordmark Oluf       3584           28.7          1.8               10.46
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03980/joss.03980/10.21105.joss.03980.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03980/joss.03980/10.21105.joss.03980.pdf) :page_facing_up: :point_left:

**Comment by Kevin-Mattheus-Moerman** [comment]
@nicoguaro, @corentin-dev this is where the review takes place. As you can see there are tickboxes at the top that will guide you through the review process. Let me know if you have any questions. Thanks again for your help!

**Comment by nicoguaro** [comment]
@PolymerGuy, I have done a first pass on the submission and have the following comments:

- There is no Statement of need on the README or the docs.
- There is a section title for an example without the example.
- There is a broken hyperlink to a reference in the README. Is this one the paper where the approach is described or is it the 2019 one?
- I think that the contribution guidelines could be more specific.

Once I know which reference to check for the approach I will continue the review.

**Comment by PolymerGuy** [comment]
@nicoguaro, thanks for the comments. I will implement the comments as soon as possible. Should I push the corrections directly to the master branch or should we use a separate revision branch for the review?

The reference in the README is a placeholder for a paper which is about to be submitted now.
For a description of the approach we use, please see the 2019 paper (https://link.springer.com/article/10.1007/s11340-019-00530-2)

**Comment by Kevin-Mattheus-Moerman** [comment]
> @nicoguaro, thanks for the comments. I will implement the comments as soon as possible. Should I push the corrections directly to the master branch or should we use a separate revision branch for the review?

You can use any branch you like this is fully up to you. As long as we compile the paper from the correct branch, e.g. `@whedon generate pdf from branch branchName`

**Comment by PolymerGuy** [comment]
@nicoguaro I have now made a new branch called JOSS-review for all changes done during the review process. Improvements based on the comments above are found in commit: 26f7c97f4efc2ce93a402c5ac8b5479523883845 .

**Comment by whedon** [comment]
:wave: @corentin-dev, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @nicoguaro, please update us on how your review is going (this is an automated reminder).

**Comment by PolymerGuy** [comment]
@corentin-dev and @nicoguaro, please let me know if there is anything I can assist you with during the review process.

**Comment by nicoguaro** [comment]
> @corentin-dev and @nicoguaro, please let me know if there is anything I can assist you with during the review process.

I'll resume activities tomorrow. So, I guess that I'll resume the review as well this week.

**Comment by nicoguaro** [comment]
@PolymerGuy, the repo (or docs) is missing a Statement of Need. Furthermore, I think that the one provided in the paper answers the question of why you wrote the software by not how other people might want to use it.

**Comment by PolymerGuy** [comment]
> @PolymerGuy, the repo (or docs) is missing a Statement of Need. Furthermore, I think that the one provided in the paper answers the question of why you wrote the software by not how other people might want to use it.

@nicoguaro  I will address this as soon as possible. Just to make sure everything is clear, the commit: [26f7c97f4efc2ce93a402c5ac8b5479523883845](https://github.com/PolymerGuy/recolo/commit/26f7c97f4efc2ce93a402c5ac8b5479523883845) did not address this in a satisfactory manner?

**Comment by nicoguaro** [comment]
> @nicoguaro I will address this as soon as possible. Just to make sure everything is clear, the commit: [26f7c97f4efc2ce93a402c5ac8b5479523883845](https://github.com/PolymerGuy/recolo/commit/26f7c97f4efc2ce93a402c5ac8b5479523883845) did not address this in a satisfactory manner?

Oh, I forgot that we have another branch for the review, sorry about that. The description is definitely much appropriate for a wider audience. The docs and the paper might benefit from being more explicit (and less technical?).

**Comment by corentin-dev** [comment]
Hi,

Sorry for the delay, concerning the software, I'm just starting to try to use it.

Please consider rebuilding the paper, no sure if it was updated by whedon.

### Installation

Most of it runs smoothly but,

- pytest does not run (at least on my machine). What I did : `python -m venv env`, `source ./env/bin/activate`, `pip install -r requirements.txt`, `pytest recolo`. I get an error: `E   ModuleNotFoundError: No module named 'natsort'`
- current tests get 37 passed, 9 warnings in 27.44s, lots of `invalid value encountered in true_divide`, can you comment please?
- maybe suggest running `python -m pytest recolo` instead?
- example `python ./examples/Experiments/impact_hammer_experiment.py` needs `SciencePlots` for some style in `matplotlib`
- a minor filename error in `bandwith_vs_grid_pitch.py`
- all examples seem to run

### Comments

The code seems to be well commented, and an automatic tool for coverage is used.

### Documentation

The documentation is automaticaly generated. The documentation is very short and should be enhanced with the way the code can be used. Most of the documentation is made of few examples. But nothing detail the structure of the code, and how to setup other cases.

### Tests

I have no idea how much percentage of the capacities are tested. Tests are present, run smoothly, with some warnings.

#### Examples

Some examples are present, even in the documentation. It can help get a hand of the software.

### Paper

> Other VFM toolkits such as PeriPyVFM are readily available but are focused on different applications.

What different application?

Can you specify the range of use of this tool? To what purpose it can be used?
Last, why `Recolo`?

Best regards,
Corentin Lothode

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03980/joss.03980/10.21105.joss.03980.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03980/joss.03980/10.21105.joss.03980.pdf) :page_facing_up: :point_left:

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf from branch JOSS-review

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS-review. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03980/joss.03980/10.21105.joss.03980.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03980/joss.03980/10.21105.joss.03980.pdf) :page_facing_up: :point_left:

**Comment by Kevin-Mattheus-Moerman** [comment]
@PolymerGuy can you address the comments above? :point_up:

**Comment by PolymerGuy** [comment]
@Kevin-Mattheus-Moerman @corentin-dev @nicoguaro  We have now implemented all comments except those regarding the documentation and the paper. We are currently working on those changes and they will be available by the end of this week.
The following changes have been made:
* The natsort-error when running pytest appears to be caused by a call to the global pytest installation. This is fixed in: https://github.com/PolymerGuy/recolo/commit/4208ee482b220a7c7c94e036733b7e24f31de101
* The warning "invalid value encountered in true_divide" is now fixed in: https://github.com/PolymerGuy/recolo/commit/f9623e026869fca1fe1be8a61e37c9deab1b96a4
* Divide by zero error was due to a test starting at zero deformation, now fixed in: https://github.com/PolymerGuy/recolo/commit/58bca9a212df2eac22958e1362b62fb3ff8f77a0
* python -m pytest recolo is now the suggested way to run the tests, fixed in:https://github.com/PolymerGuy/recolo/commit/ 4208ee482b220a7c7c94e036733b7e24f31de101
* SciencePlots is now added to requirements.txt in: https://github.com/PolymerGuy/recolo/commit/3835ea12f7f57d1ec0c6cfdf2d233736399ac9e2
* Fixed typo in filename: https://github.com/PolymerGuy/recolo/commit/4f16d39b90ddfa7427464947d746e722f68ab4e0
* I have added coverage to the suggested command for running the tests: https://github.com/PolymerGuy/recolo/commit/b22436490d0e54c78776870efd454551abe9cf4e

Regarding the documentation and the paper, we will add automatic API documentation which parses the docstrings and also provides an overview of the formatting of the package. Further, we are working on a separate section acting as an overview, meant to give the user an understanding of the overall design of this tool as well as suggestions on where to start and relevant examples. The paper will be updated according to the comments from @corentin-dev and @nicoguaro above.

**Comment by PolymerGuy** [comment]
@Kevin-Mattheus-Moerman @corentin-dev @nicoguaro We have now updated the documentation and the paper according to the comments above. For convenience, I have merged the JOSS-review branch with the master branch so that  readthedocs builds the updated documentation. The updated documentation can be found here: https://recolo.readthedocs.io/en/latest/?badge=latest

**Comment by Kevin-Mattheus-Moerman** [comment]
@nicoguaro, @corentin-dev I hope you are doing well. This is just a check in to see how things are going with the review. Are you able to pick this up and review the changes made? Or could you summarise what points still need attention? Thanks again for your help!!!

**Comment by corentin-dev** [comment]
With the changes, everything seems in order to me.

**Comment by Kevin-Mattheus-Moerman** [comment]
@nicoguaro can you also review the latest changes? Let us know if there are still major points that need work. Thanks again for your continued help!

**Comment by nicoguaro** [comment]
@Kevin-Mattheus-Moerman, @PolymerGuy,  I agree with @corentin-dev. Particularly, I think that the general description is much clearer now.

I apologize for the time it takes me for this submission!

**Comment by Kevin-Mattheus-Moerman** [comment]
@nicoguaro thanks for your help here! Do not worry we've seen all submissions slow down significantly and we've been flexible because we appreciate the efforts especially during these challenging times. 

@nicoguaro I see all your boxes are ticked, can you confirm that you are happy to recommend acceptance of this submission?

**Comment by nicoguaro** [comment]
@Kevin-Mattheus-Moerman, yes. I recommend the publication.

**Comment by Kevin-Mattheus-Moerman** [comment]
@PolymerGuy since the reviewers have recommended acceptance we will now proceed to process this submission for acceptance. 

I have proofread your paper and it looks fine. However I do have one recommendation below:
- [ ] I recommend you rephrase `Still, it is of utmost importance to provide structural engineers with a detailed knowledge of loads, understand the  underlying physics and predict how structures respond during extreme loading events.` to something like `Still, it is of the utmost importance to provide structural engineers with a detailed knowledge of loads, an understanding of the  underlying physics, and the ability to predict how structures respond during extreme loading events.`

Further to the above, please work on the following remaining steps: 
- [ ] We recommend you proofread the paper yourself once more and in particular verify that the author list is accurate and complete and that the acknowledgements are in order. 
- [ ] If you have made any changes please update the paper by calling `@editorialbot generate pdf from branch JOSS-review` (note the recent renaming of our editorial bot). 
- [ ] Please confirm what the version tag for this publication should be e.g. is it still `v1.0.0`? 
- [ ] Once the above is complete please archive a copy of the software on [ZENODO](https://zenodo.org/). You can do this manually but some find [these automated steps](https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content) useful. Once you have archived the software please report back here with the ZENODO doi. 
- [ ] For the archived version please ensure that the authors name set and order is the same and that the title of the archive also matches that of the paper. Note that you may need to manually edit this information. You may also consider adding ORCID information for authors. 

Let me know when the above is finished.

**Comment by PolymerGuy** [comment]
@editorialbot generate pdf from branch JOSS-review

**Comment by PolymerGuy** [comment]
@whedon  generate pdf from branch JOSS-review

**Comment by PolymerGuy** [comment]
@editorialbot generate pdf from branch JOSS-review

**Comment by PolymerGuy** [comment]
@Kevin-Mattheus-Moerman The paper has not been compiled after the review comments have been implemented and approved by the reviewers. All changes made during this review has now been merged to -master.

- I have now proofread the paper and confirmed the list of authors and acknowledgements.
- I tried to compile the paper using editorialbot, but the accepted commands seems to be changed.
- I have now released the software with a tag v1.0.0.
- The software has now been archived using Zenodo, using a tag v1.0.1.
- The Zenodo DOI is: https://doi.org/10.5281/zenodo.6324870
- For the list of authors, I did not find any list of authors for the archive on Zenodo?

**Comment by Kevin-Mattheus-Moerman** [comment]
@editorialbot set JOSS-review as branch

**Comment by Kevin-Mattheus-Moerman** [comment]
@editorialbot generate pdf

**Comment by Kevin-Mattheus-Moerman** [comment]
@editorialbot generate pdf

**Comment by Kevin-Mattheus-Moerman** [comment]
@PolymerGuy apologies, the editorial bot recently changed as you can see. It should work now as shown. 

1. Can you clarify some things. Did you say that the changes in the paper have not been reviewed yet? If so I'll ask the reviewers to have a quick look. 
2. On the ZENODO archive, you need to edit the title and add your other authors (like on the paper). Have you been able to edit the archive? ZENODO lets you do this.

**Comment by PolymerGuy** [comment]
@Kevin-Mattheus-Moerman The state of the papers is as it was when the reviewers replied that the amendments made were answering their comments. 
I have now been able to edit the ZENODO archive and the meta-data now matches the JOSS-paper.

**Comment by Kevin-Mattheus-Moerman** [comment]
@editorialbot set 10.5281/zenodo.6324870 as archive

**Comment by Kevin-Mattheus-Moerman** [comment]
@openjournals/dev are the archive links no longer presented as URLs (in both the return comment and the DOI listed at the top of this issue)? I think it would be better to have them be URLs to so that I can click on them to verify the link points to the right thing.

**Comment by Kevin-Mattheus-Moerman** [comment]
@editorialbot recommend-accept

**Comment by Kevin-Mattheus-Moerman** [comment]
@openjournals/dev am I doing something wrong here? :point_up:

**Comment by xuanxu** [comment]
The bib file contains references with multiple/duplicated ISBNs.
[This PR](https://github.com/PolymerGuy/recolo/pull/1) should fix it.

**Comment by Kevin-Mattheus-Moerman** [comment]
Thanks, @xuanxu did you also see my other query on the urls for DOIs? :point_up:

**Comment by Kevin-Mattheus-Moerman** [comment]
@PolymerGuy can you check that PR? :point_up:

**Comment by xuanxu** [comment]
> Thanks, @xuanxu did you also see my other query on the urls for DOIs? :point_up: 

Saw it now. Yeah we'll add links, at least in the reply comment.

**Comment by PolymerGuy** [comment]
@Kevin-Mattheus-Moerman I have now verified the changes made in the PR and ensured that the ISBN numbers are correct. The PR has been accepted and merged.

**Comment by Kevin-Mattheus-Moerman** [comment]
@editorialbot recommend-accept

**Comment by Kevin-Mattheus-Moerman** [comment]
@editorialbot accept deposit=true

**Comment by Kevin-Mattheus-Moerman** [comment]
@editorialbot accept

**Comment by Kevin-Mattheus-Moerman** [comment]
@PolymerGuy congratulations on getting this work published in JOSS! Thanks for your patience during this longer than usual review process. 

@nicoguaro, @corentin-dev thanks for your review efforts!!!
